### PR TITLE
New package: git-credential-oauth-0.8.0

### PR DESCRIPTION
**Pull request by hickford**
https://github.com/hickford/git-credential-oauth (I'm the upstream developer)

#### Testing the changes
- I tested the changes in this PR: **NO** (I don't have a Void Linux system)

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

### CODE REVIEW

File: srcpkgs/git-credential-oauth/template
Comment (hickford) [inline srcpkgs/git-credential-oauth/template:6]: Is qemu necessary? I copied from another Go template

Comment (tranzystorek-io) [reply srcpkgs/git-credential-oauth/template]: not here, it's used for the `vtargetrun` function to run, for example to run a built binary to generate shell completions

Comment (hickford) [reply srcpkgs/git-credential-oauth/template]: Thanks

File: srcpkgs/git-credential-oauth/template
Comment (tranzystorek-io) [inline srcpkgs/git-credential-oauth/template:11]:
```suggestion
distfiles="https://github.com/hickford/git-credential-oauth/archive/refs/tags/v${version}.tar.gz"
```

Comment (hickford) [reply srcpkgs/git-credential-oauth/template]: Both links work, is there any difference?

Comment (tranzystorek-io) [reply srcpkgs/git-credential-oauth/template]: Just a good practice, and there have been cases where a tag is named the same as a branch and fetch fails due to ambiguity

**Review by tranzystorek-io — APPROVED** [review]
LGTM

Tested, works correctly, but as mentioned before, a Void package maintainer generally should be able to test their package. I'd suggest someone else taking this over before it's officially merged in.

**Comment by hickford** [comment]
@tranzystorek-io I agree, best maintainer would be a Void Linux user, but I'm happy to maintain this package in the interim, as I do in other distros. Unless I'm missing something, `build_style=go` does all the work, so package update is as simple as updating the version and checksum. CI covers package build.

**Comment by tranzystorek-io** [comment]
If what we aimed for was just bumping template version on release. there are bots and scripts for that, but a maintainer is generally expected to manually run some functional tests on their package.

**Comment by sgn** [comment]
> If what we aimed for was just bumping template version on release. there are bots and scripts for that, but a maintainer is generally expected to manually run some functional tests on their package.

This!

**Pull request closed without merging**
